Golem.Tests/ProviderFailureHandlingTest.cs

```diff
@@ -0,0 +1,131 @@
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading.Channels;
+using System;
+using System.Management;
+
+using Golem.Tools;
+using Golem.Yagna.Types;
+
+using GolemLib;
+using GolemLib.Types;
+
+using Microsoft.Extensions.Logging;
+using System.Runtime.InteropServices;
+
+namespace Golem.Tests
+{
+    [Collection(nameof(SerialTestCollection))]
+    public class ErrorHandlingTests : JobsTestBase
+    {
+
+        public ErrorHandlingTests(ITestOutputHelper outputHelper, GolemFixture golemFixture)
+            : base(outputHelper, golemFixture)
+        { }
+
+        [OsFact(nameof(OSPlatform.Windows))]
+        public async Task StartTriggerErrorStop_VerifyStatusAsync()
+        {
+            // Having
+            var logger = _loggerFactory.CreateLogger(nameof(StartTriggerErrorStop_VerifyStatusAsync));
+
+            string golemPath = await PackageBuilder.BuildTestDirectory();
+            logger.LogInformation("Path: " + golemPath);
+
+            var golem = await TestUtils.Golem(golemPath, _loggerFactory);
+            var statusChannel = TestUtils.PropertyChangeChannel<Golem, GolemStatus?>((Golem)golem, nameof(Golem.Status), _loggerFactory);
+
+            var jobStatusChannel = PropertyChangeChannel<IJob, JobStatus>(null, "");
+
+            Channel<Job?> jobChannel = PropertyChangeChannel(golem, nameof(IGolem.CurrentJob), (Job? currentJob) =>
+            {
+                jobStatusChannel = PropertyChangeChannel(currentJob, nameof(currentJob.Status),
+                    (JobStatus v) => _logger.LogInformation($"Current job Status update: {v}"));
+
+            });
+
+            // Then
+            var startTask = golem.Start();
```

Comment:
This could be written much simpler. Look at `StartStop_VerifyStatusAsync` for example. In these tests you don't need to track changes but only state, so it might be useful.

Reply:
I need a channel because I have no awaitable `Task`.
Provider just gets killed and I have no indication that status already had changed.
https://github.com/golemfactory/gamerhash-facade/blob/334c5add372483982a3cf83584cb7bd029d65b40/Golem.Tests/ProviderFailureHandlingTest.cs#L71